pkg/codefresh/git-source.go

```diff
@@ -0,0 +1,94 @@
+package codefresh
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/codefresh-io/go-sdk/pkg/codefresh/model"
+)
+
+type (
+	IGitSourceAPI interface {
+		List(runtimeName string) ([]model.GitSource, error)
+	}
+
+	gitSource struct {
+		codefresh *codefresh
+	}
+
+	graphQlGitSourcesListResponse struct {
+		Data struct {
+			GitSources model.GitSourcePage
+		}
+		Errors []graphqlError
+	}
+)
+
+func newGitSourceAPI(codefresh *codefresh) IGitSourceAPI {
+	return &gitSource{codefresh: codefresh}
+}
+
+func (g *gitSource) List(runtimeName string) ([]model.GitSource, error) {
+	jsonData := map[string]interface{}{
+		"query": `{
+			gitSources {
+			  edges {
+				node {
+				  metadata {
+					name
+				  }
+				  self {
+					path
+					repoURL
+					revision
+					size
+					status {
+					  syncStatus
+					  syncStartedAt
+					  syncFinishedAt
+					  healthStatus
+					  revision
+					}
+				  }
+				}
+			  }
+			}
+		  }`,
+	}
+
+	response, err := g.codefresh.requestAPI(&requestOptions{
+		method: "POST",
+		path:   "/argo/api/graphql",
```

Comment:
should be `/2.0/api/garphql`
I will later refactor it all into a unified graphql func that also checks the response return code, so we will have better error messages. But for now - just the path fix is important.